packages/dom/README.md

```diff
@@ -204,7 +204,7 @@ and has a valueAsNumber
 
 _Parameters_
 
--   _element_ `HTMLElement`: The HTML element.
+-   _element_ `Element`: The HTML element.
```

Comment:
It seems like `HTMLElement` is a narrow type than is necessary. From the documentation on `Element`:

> `Element` is the most general base class from which all objects in a `Document` inherit. It only has methods and properties common to all kinds of elements. More specific classes inherit from `Element`

Whereas `HTMLElement` is:

> Any HTML element. Some elements directly implement this interface, while others implement it via an interface that inherits it.

Maybe it should be left as `HTMLElement` instead? 🤔 I'm not sure 😬 I'll defer to @sirreal again on this one, sorry to pull you back in Jon!

Reply:
AS far as I know, `Element` can also be `SVGElement` and perhaps custom types. Just `Element` sounds good though. 🤷‍♀️

Reply:
I'd say it's good practice to generalize parameter types as much as possible. That will limit what we can do with the values internally. Externally, it will allow these functions to be used in more places.

For example, this function checks the [`nodeName` property](https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeName), which exists on [`Node`](https://developer.mozilla.org/en-US/docs/Web/API/Node) ([from which Element inherits](https://developer.mozilla.org/en-US/docs/Web/API/Element)):

```ts
export default function isHTMLInputElement( node: Node ): node is HTMLInputElement {
	return node.nodeName === 'INPUT';
}
```

That seems ideal to me. The parameter is exactly as strict as it must be for it to be used internally.